### PR TITLE
feat(ts): support Cloudflare Workers + refactor build pipeline with `tsdown` + introduce e2e tests

### PR DESCRIPTION
**Pull request by jkomyno**
This PR:
- closes [PLEN-1071](https://linear.app/composio/issue/PLEN-1071/support-for-cloudfare-workers)
- adds **Cloudflare Workers runtime (`workerd`)** compatibility to `@composio/core`
  - every functionality is exposed, except `composio.files.*` (`composio.files.upload()`, `composio.files.download()`)
- introduces a comprehensive **E2E testing infrastructure** across multiple JavaScript runtimes
- refactors the **build pipeline from `tsup` to `tsdown`** with shared configuration

---

## Cloudflare Workers Support

After deep investigation and multiple attempts, I achieved Cloudflare Workers compatibility by:

### Root Cause

The original `@composio/core` had `require()` calls that, after `pnpm build`, resulted in:
```js
import { createRequire } from 'node:module';
const path = createRequire(import.meta.url);
```

This yielded cryptic errors in Cloudflare Workers:
```
Uncaught TypeError: The argument 'path' The argument must be a file URL object, a file URL string, or an absolute path string.. Received 'undefined'
```

### Solution: Platform Abstraction

Introduced a **platform-dependent abstraction layer** via [subpath imports](https://nodejs.org/api/packages.html#subpath-imports):

- `ts/packages/core/src/platform/types.ts`: `Platform` interface for file system and OS operations
- `ts/packages/core/src/platform/node.ts`: Node.js implementation (full FS support)
- `ts/packages/core/src/platform/workerd.ts`: Cloudflare Workers implementation (no-op stubs)

The `#platform` import alias resolves to the correct implementation based on runtime:
```json
"imports": {
  "#platform": {
    "workerd": "./dist/platform/workerd.mjs",
    "edge-light": "./dist/platform/workerd.mjs",
    "node": "./dist/platform/node.mjs",
    "default": "./dist/platform/node.mjs"
  }
}
```

### Additional Fixes

- **Removed `uuid` dependency** — replaced with native `crypto.randomUUID()` to fix CommonJS compatibility
- **`node_compat` flag no longer needed** — at least for the base scenario tested in `pnpm --filter "@test-e2e/cf-workers-basic" test:e2e`

---

## E2E Testing Infrastructure

Introduced `ts/e2e-tests/`: a structured test suite for validating `@composio/core` across different runtimes:

```
ts/e2e-tests/
├── _utils/                           # Shared test infrastructure
│   ├── Dockerfile.node               # Docker image for Node.js tests
│   └── run-docker-test.sh            # Generic Docker test runner
└── runtimes/
    ├── node/                         # Node.js runtime tests
    │   ├── cjs-basic/                # CommonJS compatibility
    │   └── esm-basic/                # ESM compatibility
    └── cloudflare/                   # Cloudflare runtime tests
        └── cf-workers-basic/         # Cloudflare Workers + vitest
```

### Commands

| Command | Description |
|---------|-------------|
| `pnpm test:e2e` | Run all E2E tests |
| `pnpm test:e2e:node` | Run Node.js tests in Docker (CJS + ESM) |
| `pnpm test:e2e:cloudflare` | Run Cloudflare Workers tests via vitest |

### CI Workflow

Added `.github/workflows/ts.test-e2e.yml`:

- **Node.js matrix**: `20.18.0`, `20.19.0`, `22.12.0` run on Docker
  - `20.18.0` may throw `ERR_REQUIRE_ESM`, useful edge case
  - `20.19.0` supports ESM by default
- **Cloudflare Workers**: relies on `wrangler@catalog:` and vitest

---

### tsup → tsdown Migration

- Replaced outdated `tsup` with `tsdown` across all packages
- Created `tsdown.config.base.ts`: a shared configuration inherited by all packages:
  - **Dual format output**: ESM (`.mjs`) + CJS (`.cjs`)
  - **TypeScript declarations**: `.d.mts` and `.d.cts`
  - **ES2022 target**
  - **External dependencies**: `zod`, `node:*` builtins

### Package Quality Checks

Integrated build-time validation via `tsdown`:
- [`@arethetypeswrong/cli`](http://npmjs.com/package/@arethetypeswrong/cli): validates TypeScript exports
- [`publint`](http://npmjs.com/package/publint): catches common publishing mistakes

### Fixed `exports` Maps

All `package.json` exports fields now correctly specify:
- Separate `import`/`require` conditions
- Proper `types` fields for each format
- Runtime-conditional exports for `@composio/core/platform`

---

## Other Changes

- **CJS support recovered** — ensured CommonJS consumers work correctly
- **`turbo.json` → `turbo.jsonc`** — enables comments in Turborepo config
- **Added `@composio/core/generated` stubs** — fixes publint validation

### CODE REVIEW

File: ts/packages/core/src/utils/fileUtils.ts [hunk 1]
```diff
@@ -121,7 +126,7 @@ const readFileContentFromURL = async (
   // Extract clean filename from URL, removing query parameters
   const url = new URL(path);
   const pathname = url.pathname;
-  let fileName = pathModule.basename(pathname);
```
[the remark this answers is not in the document]
Comment (jkomyno) [reply ts/packages/core/src/utils/fileUtils.ts:124]: https://github.com/ComposioHQ/composio/pull/2360/commits/4be46e594d0a0fe47be2f919e44fb7660fce2f2d

**Review by haxzie — APPROVED** [review]
